src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp

```diff
@@ -859,8 +868,161 @@ void CompositionalMultiphaseBase::initializePostInitialConditionsPreSubGroups()
+    arrayView1d< real64 const > const refPorosity = solid.getReferencePorosity();
+    arrayView2d< real64 const > const porosity = solid.getPorosity();
+
+    integer numPhases = m_numPhases;
+    forAll< parallelDevicePolicy<> >( subRegion.size(), [=] GEOSX_HOST_DEVICE ( localIndex const ei )
+    {
+      if( elemGhostRank[ei] >= 0 )
+      {
+        return;
+      }
+
+      // To match our "reference", we have to use reference porosity here, not the actual porosity when we compute averages
+      real64 const uncompactedPoreVolume = volume[ei] * refPorosity[ei];
+      real64 const dynamicPoreVolume = volume[ei] * porosity[ei][0];
+
+      subRegionAvgPressureNumerator += uncompactedPoreVolume * pres[ei];
+      subRegionMaxPressure.max( pres[ei] );
+      subRegionMinPressure.min( pres[ei] );
+
+      subRegionAvgTemperatureNumerator += uncompactedPoreVolume * temp[ei]; // for real thermal change pore volume to volume
+      subRegionMaxTemperature.max( temp[ei] );
+      subRegionMinTemperature.min( temp[ei] );
+
+      subRegionTotalUncompactedPoreVolume += uncompactedPoreVolume;
+      for( integer ip = 0; ip < numPhases; ++ip )
+      {
+        subRegionPhaseDynamicPoreVolume[ip] += dynamicPoreVolume * phaseVolFrac[ei][ip];
+      }
+    } );
+
+    m_averagePressure += subRegionAvgPressureNumerator.get();
+    if( subRegionMinPressure.get() < m_minimumPressure )
+    {
+      m_minimumPressure = subRegionMinPressure.get();
+    }
+    if( subRegionMaxPressure.get() > m_maximumPressure )
+    {
+      m_maximumPressure = subRegionMaxPressure.get();
+    }
+
+    m_averageTemperature += subRegionAvgTemperatureNumerator.get();
+    if( subRegionMinTemperature.get() < m_minimumTemperature )
+    {
+      m_minimumTemperature = subRegionMinTemperature.get();
+    }
+    if( subRegionMaxTemperature.get() > m_maximumTemperature )
+    {
+      m_maximumTemperature = subRegionMaxTemperature.get();
+    }
+
+    totalUncompactedPoreVolume += subRegionTotalUncompactedPoreVolume.get();
+    for( integer ip = 0; ip < m_numPhases; ++ip )
+    {
+      m_phasePoreVolume[ip] += subRegionPhaseDynamicPoreVolume[ip].get();
+    }
+
+  } );
+
+  m_averagePressure = MpiWrapper::sum( m_averagePressure );
+  m_minimumPressure = MpiWrapper::min( m_minimumPressure );
+  m_maximumPressure = MpiWrapper::max( m_maximumPressure );
+
+  m_averageTemperature = MpiWrapper::sum( m_averageTemperature );
+  m_minimumTemperature = MpiWrapper::min( m_minimumTemperature );
+  m_maximumTemperature = MpiWrapper::max( m_maximumTemperature );
+
+  m_totalPoreVolume = 0.0;
+  totalUncompactedPoreVolume = MpiWrapper::sum( totalUncompactedPoreVolume );
+  for( integer ip = 0; ip < m_numPhases; ++ip )
+  {
+    m_phasePoreVolume[ip] = MpiWrapper::sum( m_phasePoreVolume[ip] );
+    m_totalPoreVolume += m_phasePoreVolume[ip];
+  }
+
+  m_averagePressure /= totalUncompactedPoreVolume;
+  m_averageTemperature /= totalUncompactedPoreVolume;
+
+  if( outputToTerminal )
```

Comment:
Are you computing all the above even if you do not display?
Are you using the information somewhere else?
If  possible, I would make this computation iff `outputToTerminal` is `true`.

Reply:
Same here, the information will be always computed when `computeStatistics` is `true`, but only accessible through `TimeHistory`